onnxruntime/core/providers/dml/DmlExecutionProvider/src/BucketizedBufferAllocator.cpp

```diff
@@ -244,17 +244,12 @@ namespace Dml
 
     void* CPUAllocator::Alloc(size_t size)
     {
-        if (size <= 0)
```

Comment:
Sorry I don't understand this code. If the pointers returned by this function are for DML internal uses only(the pointers will not be passed to the core framework or mlas), then you are free to use any malloc function, you don't need to use onnxruntime::AllocatorDefaultAlloc. Otherwise, when the pointer is passed outside, who will free it, and how? Supposedly only DML code has access to this BucketizedBufferAllocator's free function.